lib/livebook/runtime/evaluator.ex

```diff
@@ -454,6 +455,26 @@ defmodule Livebook.Runtime.Evaluator do
   defp code_error?(%CompileError{}), do: true
   defp code_error?(_error), do: false
 
+  defp reorder_binding(binding, prev_binding) do
+    # We keep the order of existing binding entries and move
+    # the new ones to the beginning, ordered alphabetically
+
+    binding_map = Map.new(binding)
+
+    unchanged_binding =
+      Enum.filter(prev_binding, fn {key, prev_val} ->
+        val = binding_map[key]
+        :erts_debug.same(val, prev_val)
+      end)
+
+    unchanged_keys = for {key, _} <- unchanged_binding, into: MapSet.new(), do: key
+
+    binding
+    |> Enum.reject(fn {key, _} -> MapSet.member?(unchanged_keys, key) end)
+    |> Enum.sort()
```

Comment:
```suggestion
    unchanged_binding
    |> Enum.reduce(binding_map, fn {key, _}, acc -> Map.delete(acc, key) end)
    |> Map.to_list()
```

Then you can skip the map_set given the binding is already a map with fast lookups.

Reply:
Beautiful!